tests/rkt_image_dependencies_test.go

```diff
@@ -0,0 +1,220 @@
+	serverURL.Value.Set("")
+
+	baseImage := getInspectImagePath()
+	_ = importImageAndFetchHash(t, ctx, baseImage)
+	emptyImage := getEmptyImagePath()
+	fileSet := make(map[string]string)
+
+	// Scenario from https://github.com/coreos/rkt/issues/1752#issue-117121841
+	//
+	// A->B
+	// A->C
+	// A->D
+	//
+	// B: prefetched
+	//
+	// C->B
+	// C->E
+	//
+	// D->B
+	// D->E
+
+	topImage := "localhost/image-a"
+	imageList := []struct {
+		shortName string
+		imageName string
+		deps      string
+		version   string
+		prefetch  bool
+
+		manifest string
+		fileName string
+	}{
+		{
+			shortName: "a",
+			imageName: topImage,
+			deps:      `{"imageName":"localhost/image-b"}, {"imageName":"localhost/image-c"}, {"imageName":"localhost/image-d"}`,
+			version:   "1",
+		},
+		{
+			shortName: "b",
+			imageName: "localhost/image-b",
+			deps:      ``,
+			version:   "1",
+			prefetch:  true,
+		},
+		{
+			shortName: "c",
+			imageName: "localhost/image-c",
+			deps:      `{"imageName":"localhost/image-b"}, {"imageName":"localhost/image-e", "labels": [{"name": "version", "value": "1"}]}`,
+			version:   "1",
+		},
+		{
+			shortName: "d",
+			imageName: "localhost/image-d",
+			deps:      `{"imageName":"localhost/image-b"}, {"imageName":"localhost/image-e", "labels": [{"name": "version", "value": "1"}]}`,
+			version:   "1",
+		},
+		{
+			shortName: "e",
+			imageName: "localhost/image-e",
+			deps:      `{"imageName":"coreos.com/rkt-inspect"}`,
+			version:   "1",
+		},
+	}
+
+	for i, _ := range imageList {
+		// We need a reference rather than a new copy from "range"
+		// because we modify the content
+		img := &imageList[i]
+
+		img.manifest = manifestDepsTemplate
+		img.manifest = strings.Replace(img.manifest, "IMG_NAME", img.imageName, -1)
+		img.manifest = strings.Replace(img.manifest, "DEPENDENCIES", img.deps, -1)
+		img.manifest = strings.Replace(img.manifest, "VERSION", img.version, -1)
+
+		tmpManifest, err := ioutil.TempFile(tmpDir, "manifest-"+img.shortName+"-")
+		if err != nil {
+			panic(fmt.Sprintf("Cannot create temp manifest: %v", err))
+		}
+		if err := ioutil.WriteFile(tmpManifest.Name(), []byte(img.manifest), 0600); err != nil {
+			panic(fmt.Sprintf("Cannot write to temp manifest: %v", err))
+		}
+		defer os.Remove(tmpManifest.Name())
+
+		baseName := "image-" + img.shortName + ".aci"
+		img.fileName = patchACI(emptyImage, baseName, "--manifest", tmpManifest.Name())
+		defer os.Remove(img.fileName)
+		fileSet[baseName] = img.fileName
+	}
+
+	server.UpdateFileSet(fileSet)
+
+	for i := len(imageList) - 1; i >= 0; i-- {
+		img := imageList[i]
+		if img.prefetch {
+			t.Logf("Importing image %q: %q", img.imageName, img.fileName)
+			testImageShortHash := importImageAndFetchHash(t, ctx, img.fileName)
+			t.Logf("Imported image %q: %s", img.imageName, testImageShortHash)
+		}
+	}
+
+	runCmd := fmt.Sprintf("%s --debug --insecure-options=image,tls run %s", ctx.Cmd(), "localhost/image-a")
```

Comment:
Please use the `topImage` variable, instead of writing the `"localhost/image-a"` again.